src/webui/webui.cpp

```diff
@@ -117,9 +117,10 @@ void WebUI::configure()
                     .arg(serverAddressString).arg(port).arg(m_httpServer->errorString());
                 LogMsg(errorMsg, Log::CRITICAL);
                 qCritical() << errorMsg;
-
+#ifdef DISABLE_GUI
```

Comment:
This is contrary to the current architecture. WebUI part should know nothing about GUI. You could emit some signal with message as parameter and handle it in Applications class.

Reply:
It is complicated because the error happens in the class constructor. So connecting after the object is created is too late.
I don't how to handle it in the current state.

Reply:
> Doesn't this work (line 119)?
> `qCritical("%s", qUtf8Printable(errorMsg));`

It only prints into logs, but not into stderr.

Reply:
Hmm, I don't know then, it does nothing for me.

Reply:
> So I don't know how to handle it in this situation.

All you need is to expand the current handling a bit by adding `QString WebUI::error() const` and storing error message in `m_errorMsg` instead of local variable. The rest of the logic is already there:
```c++
m_webui = new WebUI(this);
#ifdef DISABLE_GUI
if (m_webui->isErrored())
    // print m_webui->error()
#endif

Reply:
> But it feels kinda dirty and only stores the last error.

This is one of the common practices. I suggested this only because it has already been implemented (with the exception of the aforementioned property `WebUI::error()`).
>only stores the last error

It stores "current" error because of what WebUI is in errored state currently (i.e. `WebUI::isErrord()` returns `true`).

Reply:
Now I'm thinking, why we want `#ifdef DISABLE_GUI` like at all?

As this code branch is no longer fatal, shouldn't it be useful to see the messages for GUI version too, if it was launched from the terminal?

Reply:
> As this code branch is no longer fatal, shouldn't it be useful to see the messages for GUI version too, if it was launched from the terminal?
> I see no downsides with it.

Anyway it's better to let intermediate layer (Application class) to decide it.

Reply:
Oops, I didn't noticed

https://github.com/qbittorrent/qBittorrent/blob/f29bbf28ccdd8b8fe6e7f97aed81e536ee4c2634/src/app/application.cpp#L34-L36

But printing still works somehow 😅

---

Anyway, I think to revert `#ifdef DISABLE_GUI` back. Because there are too many places where it is used, like

https://github.com/qbittorrent/qBittorrent/blob/f29bbf28ccdd8b8fe6e7f97aed81e536ee4c2634/src/app/application.cpp#L780-L783

And it looks out of place.